pkg/controller/operators/catalog/operator.go

```diff
@@ -564,28 +580,37 @@ func (o *Operator) syncResolvingNamespace(obj interface{}) error {
 	logger.Debug("resolving subscriptions in namespace")
 
 	// resolve a set of steps to apply to a cluster, a set of subscriptions to create/update, and any errors
-	steps, subs, err := o.resolver.ResolveSteps(namespace, resolver.NewNamespaceSourceQuerier(resolverSources))
+	steps, updatedSubs, err := o.resolver.ResolveSteps(namespace, resolver.NewNamespaceSourceQuerier(resolverSources))
 	if err != nil {
 		return err
 	}
 
-	// any subscription in the namespace with manual approval will force generated installplans to be manual
-	// TODO: this is an odd artifact of the older resolver, and will probably confuse users. approval mode could be on the operatorgroup?
-	installPlanApproval := v1alpha1.ApprovalAutomatic
-	for _, sub := range subs {
-		if sub.Spec.InstallPlanApproval == v1alpha1.ApprovalManual {
-			installPlanApproval = v1alpha1.ApprovalManual
-			break
+	// create installplan if anything updated
+	if len(updatedSubs) > 0 {
+		logger.Debug("resolution caused subscription changes, creating installplan")
+		// any subscription in the namespace with manual approval will force generated installplans to be manual
+		// TODO: this is an odd artifact of the older resolver, and will probably confuse users. approval mode could be on the operatorgroup?
+		installPlanApproval := v1alpha1.ApprovalAutomatic
+		for _, sub := range subs {
+			if sub.Spec.InstallPlanApproval == v1alpha1.ApprovalManual {
+				installPlanApproval = v1alpha1.ApprovalManual
+				break
+			}
 		}
-	}
-	installplanReference, err := o.createInstallPlan(namespace, subs, installPlanApproval, steps)
-	if err != nil {
-		logger.WithError(err).Debug("error creating installplan")
-		return err
+		installPlanReference, err := o.createInstallPlan(namespace, subs, installPlanApproval, steps)
+		if err != nil {
+			logger.WithError(err).Debug("error creating installplan")
+			return err
+		}
+		if err := o.updateSubscriptionStatus(namespace, updatedSubs, installPlanReference); err != nil {
+			logger.WithError(err).Debug("error ensuring subscription installplan state")
+			return err
+		}
+		return nil
 	}
 
-	if err := o.updateSubscriptionSetInstallPlanState(namespace, subs, installplanReference); err != nil {
-		logger.WithError(err).Debug("error ensuring subscription installplan state")
+	if err := o.updateSubscriptionStatus(namespace, subs, nil); err != nil {
```

Comment:
What case does this cover? It's not immediately apparent to me.

Reply:
this is ensuring that subscriptions get their last update time updated whenever we've fully processed them, so that on a resync we see the time as newer and don't process when we don't need to